From d804614065ea0728e7374257a3552582e51bf6b0 Mon Sep 17 00:00:00 2001
From: Mathieu Bivert <mathieu.bivert@intel.com>
Date: Tue, 23 Apr 2013 10:49:06 +0200
Subject: [PORT FROM MAIN] crashtool: simplify kernel API (kct.h)

BZ: 100456

Repercut some changes on userland header (useless flag removal; moving
flag to filter event so that it can be used in kernel).

Change-Id: I242e88b8d6e737785503b916d800fcdeb20c0964
Orig-Change-Id: I1ddce95d14c0b81a9d5f8039ce7fdb1c451700dc
Signed-off-by: Mathieu Bivert <mathieu.bivert@intel.com>
Reviewed-on: http://android.intel.com:8080/104051
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Tested-by: Zurmely, PierreX <pierrex.zurmely@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 libc/include/ctm.h             |    2 +-
 libc/include/lct.h             |    1 -
 libc/kernel/common/linux/kct.h |    3 +++
 3 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/libc/include/ctm.h b/libc/include/ctm.h
index d01cee7..a45611b 100644
--- a/libc/include/ctm.h
+++ b/libc/include/ctm.h
@@ -47,7 +47,7 @@ typedef int (*ev_handler_t)(struct ct_event *ev);
 extern int ctm_nl_init(void);
 extern int ctm_nl_exit(int fd);
 extern int ctm_nl_sendto_kct(int fd, int type, const void *data, unsigned int size);
-extern struct kct_packet *ctm_nl_get_packet(int fd, int flags);
+extern struct kct_packet *ctm_nl_get_packet(int fd);
 extern int ctm_nl_dclr_pid(int fd);
 
 /* ctm_comm.c */
diff --git a/libc/include/lct.h b/libc/include/lct.h
index 0cbf65c..0748304 100644
--- a/libc/include/lct.h
+++ b/libc/include/lct.h
@@ -5,7 +5,6 @@
 #include <sys/un.h>
 #include <linux/kct.h>
 
-#define	EV_FLAGS_PRIORITY_LOW	(1<<0)
 /* prefix of properties used to filter events */
 #define	PROP_PREFIX "dev.log."
 
diff --git a/libc/kernel/common/linux/kct.h b/libc/kernel/common/linux/kct.h
index 9264971..a9eb766 100644
--- a/libc/kernel/common/linux/kct.h
+++ b/libc/kernel/common/linux/kct.h
@@ -3,6 +3,9 @@
 
 #  include <linux/netlink.h>
 
+/* flags to optionally filter events on android property activation */
+#define	EV_FLAGS_PRIORITY_LOW	1
+
 #  ifndef MAX_SB_N
 #    define MAX_SB_N 32
 #  endif
-- 
1.7.4.1

